File: fc-chat/demo/part3/ch3/03-2-message-system-add-channel/src/main/java/net/prostars/messagesystem/dto/websocket/inbound/InviteRequest.java
package net.prostars.messagesystem.dto.websocket.inbound;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import net.prostars.messagesystem.constant.MessageType;
import net.prostars.messagesystem.dto.domain.InviteCode;

public class InviteRequest extends BaseRequest {

  private final InviteCode userInviteCode;

  @JsonCreator
  public InviteRequest(@JsonProperty("userInviteCode") InviteCode userInviteCode) {
    super(MessageType.INVITE_REQUEST);
    this.userInviteCode = userInviteCode;
  }

  public InviteCode getUserInviteCode() {
    return userInviteCode;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/inbound/InviteRequest.kt
package com.yscorp.withpush.messagesystem.dto.websocket.inbound

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.yscorp.withpush.messagesystem.constant.MessageType
import com.yscorp.withpush.messagesystem.dto.domain.InviteCode


class InviteRequest @JsonCreator constructor(@JsonProperty("userInviteCode") val userInviteCode: InviteCode) :
    BaseRequest(MessageType.INVITE_REQUEST) {
}
